components/CardForm/index.js

```diff
@@ -68,7 +100,12 @@ export default function CardForm({
   );
 
   const handleCategoryChange = (event) => {
-    setSelectedCategory(event.target.value);
+    setSelectedCategory((prevCategory) => {
+      const newCategory = event.target.value;
+      if (newCategory) {
+        return newCategory;
+      }
+    });
```

Comment:
Again, this shouldn't change the functionality, but it could be argued that it is clearer what is going on. 

```suggestion
    const newCategory = event.target.value;
    if (newCategory) {
      setSelectedCategory(newCategory)
    }
```

Reply:
I tried the suggested change, but like that the color changes after the second selection and not directly by first selection. Don't know why.